astropy/modeling/tabular.py

```diff
@@ -0,0 +1,280 @@
+
+    _id = 0
+
+    def __init__(self, points=None, lookup_table=None, method='linear',
+                 bounds_error=True, fill_value=np.nan, **kwargs):
+
+        n_models = kwargs.get('n_models', 1)
+        if n_models > 1:
+            raise NotImplementedError('Only n_models=1 is supported.')
+        super(Tabular, self).__init__(**kwargs)
+
+        if lookup_table is not None:
+            lookup_table = np.asarray(lookup_table)
+            if self.lookup_table.ndim != lookup_table.ndim:
+                raise ValueError("lookup_table should be an array with "
+                                 "{0} dimensions".format(self.lookup_table.ndim))
+            self.lookup_table = lookup_table
+        if points is None:
+            self._points = tuple(np.arange(x, dtype=np.float)
+                                 for x in self.lookup_table.shape)
+        else:
+            if self.lookup_table.ndim == 1:
+                self._points = (points,)
+            else:
+                self._points = points
+            if len(self._points) != self.lookup_table.ndim:
+                raise ValueError("Expected grid points in "
+                                 "{0} directions, got {1}".format(self.lookup_table.ndim,
+                                                                  len(self._points)))
+
+        self.bounds_error = bounds_error
+        self.method = method
+        self.fill_value = fill_value
+
+    @property
+    def points(self):
+        """
+        An iterable of points defining the values in the lookup table.
+        """
+        return self._points
+
+    @property
+    def bounding_box(self):
+        """
+        Tuple defining the default ``bounding_box`` limits,
+        ``(points_low, points_high)``.
+
+        Examples
+        --------
+        >>> from astropy.modeling.models import Tabular1D, Tabular2D
+        >>> t1 = Tabular1D(points=[1,2,3], lookup_table=[10, 20, 30])
+        >>> t1.bounding_box
+        (1, 3)
+        >>> t2 = Tabular2D(points=[[1,2,3],[2,3,4]], lookup_table=[[10,20,30],[20,30,40]])
+        >>> t2.bounding_box
+        ((2, 4), (1, 3))
+
+        """
+        bbox = [(min(p), max(p)) for p in self.points][::-1]
+        if len(bbox) == 1:
+            bbox = bbox[0]
+        return tuple(bbox)
+
+    def evaluate(self, *inputs):
+        """
+        Return the interpolated values at the input coordinates.
+
+        Parameters
+        ----------
+        inputs : list of scalars or ndarrays
+            Input coordinates. The number of inputs must be equal
+            to the dimensions of the lookup table.
+        """
+        inputs = np.array(inputs[: self.n_inputs]).T
+        if not has_scipy:
+            raise ImportError("This model requires scipy >= v0.14")
+        return interpn(self.points, self.lookup_table, inputs,
+                       method=self.method, bounds_error=self.bounds_error,
+                       fill_value=self.fill_value)
+
```

Comment:
Add `__repr__()` and `__str__()` to expose `points` and `lookup_table` in info printouts/strings? (Please see my prior comment on this matter.)